scripts/generate_tilesets.py

```diff
@@ -197,28 +197,33 @@ def assert_year(img_src, year, tiles_gdf):
         tiles_gdf (GeoDataframe): tiles geodataframe
     """
 
-    try:
-        assert (year=='multi-year' and 'year_tile' in tiles_gdf.keys()) or (str(year).isnumeric() and 'year_tile' not in tiles_gdf.keys() and img_src!='WMS' and img_src!='MIL') 
-    except:
+    if img_src=='XYZ' or img_src=='FOLDER':
         if year=='multi-year':
-            logger.error("Option 'multi-year' chosen but the tile geodataframe does not contain a 'year' column." 
-                        "Please add it while producing the tile geodtaframe or set a numeric year in the configuration file.")
-            sys.exit(1)
-        elif year:
-            if img_src=='WMS' or img_src=='MIL':
-                logger.error("The connectors WMS and MIL do not support year information. The input year will be ignored.") 
+            if 'year_tile' in tiles_gdf.keys():
+                pass
             else:
-                logger.error("Option 'year' chosen but the tile geodataframe contains a 'year' column." 
-                            "Please delete it while producing the tile geodtaframe or set the 'multi-year' option in the configuration file.")
+                logger.error("Option 'multi-year' chosen but the tile geodataframe does not contain a 'year' column. " 
+                "Please add it while producing the tile geodataframe or set a numeric year in the configuration file.")
                 sys.exit(1)
-        elif 'year_tile' in tiles_gdf.keys():
-            if img_src=='WMS' or img_src=='MIL':
-                logger.error("The connectors WMS and MIL do not support year information. Please provide a tile geodataframe without it.")            
+        elif str(year).isnumeric():
+            if 'year_tile' not in tiles_gdf.keys():
+                pass
             else:
+                logger.error("Option 'year' chosen but the tile geodataframe contains a 'year' column. " 
+                "Please delete it while producing the tile geodataframe or set the 'multi-year' option in the configuration file.")
+                sys.exit(1)
```

Comment:
If we want to be perfectly correct here, we should avoid negative condition as much as possible. In addition, it would allow to shorten the code.

```
elif str(year).isnumeric() & ('year_tile' in tiles_gdf.keys():
     logger.error("...")
     sys.exit(1)
```

Reply:
I did it like that trying to making explicit all the option but yes we can remove some lines.